.github/workflows/doc-publish.yml

```diff
@@ -9,24 +9,31 @@ permissions:
   contents: write
 
 jobs:
-  build-docs:
-    uses: ./.github/workflows/doc-build.yml
+  build-sphinx-docs:
+    uses: ./.github/workflows/sphinx-doc-build.yml
 
   publish-docs:
     runs-on: ubuntu-latest
-    needs: build-docs
+    needs: build-sphinx-docs
     steps:
-      - name: Download build documentation
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Download Sphinx documentation
         uses: actions/download-artifact@v4
         with:
-          name: docs
+          name: sphinx_docs
+          path: ./doc/github_pages
 
-      - name: Unzip
-        run: unzip docs.zip
+      - name: Unzip Sphinx documentation
+        working-directory: ./doc/github_pages
+        run: |
+          unzip sphinx_docs.zip
+          rm sphinx_docs.zip
 
       - name: Deploy to GitHub Pages
         uses: peaceiris/actions-gh-pages@v4
         with:
           github_token: ${{ secrets.GITHUB_TOKEN }}
-          publish_dir: docs
+          publish_dir: doc/github_pages
```

Comment:
Why did you decide to add _github_pages_ inside of _doc_ instead of renaming _doc_ to _githhub_pages_?

Reply:
That's the `doc` directory of the git repo, I had to add steps to `doc-publish.yml` to checkout the git repo, the new html file is in the repo at `doc/github_pages/index.html`. So it seemed easier to unzip stuff in subdirs from there (eg. `doc/github_pages/sphinx_docs/`) and publish everything in `doc/github_pages/`. Other future stuff can be unzipped in other subdirs, and if there's anything else static in future it can be checked into the repo under `doc/github_pages/` and it will get published.

Reply:
Ah great!